src/uwtools/config.py

```diff
@@ -80,6 +80,35 @@ def dump_file(self, output_path):
         ''' Interface to write a config object to a file at the
         output_path provided. '''
 
+    @abc.abstractmethod
+    def compare_config(self, user_dict, base_dict=None):
+        #pylint: disable=multiple-statements
+        '''
+        Assuming a section, key/value structure of configuration types,
+        compare the dictionary to the values stored in the external file.
+        '''
+        if base_dict is None:
+            base_dict = self.data
+
+        mishdr = {}
```

Comment:
It's unclear what `mis` is here, it may also be easier to understand if you spell out `header` and `value`.

Reply:
Habit of keeping short names, I'll spell it out and add a comment here, thanks

Reply:
Variable names extended!